SiteParser.java

```diff
@@ -0,0 +1,307 @@
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import org.glassfish.jersey.servlet.ServletContainer;
+import org.jsoup.Connection;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import javax.imageio.ImageIO;
+import javax.ws.rs.core.Response;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.net.*;
+import java.nio.charset.Charset;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static au.com.bytecode.opencsv.CSVWriter.NO_QUOTE_CHARACTER;
+
+/**
+ * Grishko
+ * 01.04.2018
+ */
+public class SiteParser extends ServletContainer {
+
+    private final static String siteUrlForParsing = "https://service.nalog.ru/tabak.do";
+    private final static String captchaFileName = "captchaImage.jpg";
+    private static String date = ""; // example = 29.03.2018 if value "" that use current date
+    private static Connection connect;
+    public final static String resourceDirectory = InitParams.getInstance().getServletConfig().getInitParameter("resourceDirectory");
+    public final static String resultDirectory = InitParams.getInstance().getServletConfig().getInitParameter("resultDirectory");
+
+    public void parse() throws IOException {
+        deletePrevJpgAndCsvFiles();
+
+        Connection.Response res = getResponseFromSite();
+        if (res != null)
+            prepareRequest(res);
+    }
+
+    public SiteParser() {
+        connect = Jsoup.connect(siteUrlForParsing).data("c", "search");
+        if (CookieHandler.getDefault() == null) {
+            CookieManager cookieManager = new CookieManager(null, CookiePolicy.ACCEPT_ALL);
+            CookieHandler.setDefault(cookieManager);
+        }
+    }
+
+    public static Response.ResponseBuilder getDataFromSite(String captchaCode) {
+        int num = captchaCode.indexOf("=");
+        String capSubstr = captchaCode.substring(num + 1, captchaCode.length());
+
+        if (date.equals("")) date = Utils.getCurrentDate();
+
+        connect.data("product", "")
+                .data("dt", date)
+                .timeout(0)
+                .data("cap", capSubstr);
+
+        try {
+            Document doc = sendRequestToSite(connect);
+            if (doc != null) {
+                return parseResponseFromSite(doc);
+            } else
+                return Response.serverError();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return Response.serverError();
+        }
+    }
+
+    private static final String NAME = "Название";
+    private static final String ADRESS = "Адрес";
+    private static final String INN = "ИНН";
+    private static final String KPP = "КПП";
```

Comment:
Here you have UPPERCASE constant. Also it's a good practice to put all constant in the top of class definition